src/scram_sha256_fmt_plug.c

```diff
@@ -0,0 +1,306 @@
+/*
+ * This software is Copyright (c) 2025, Dhiru Kholia <dhiru.kholia at gmail.com>,
+ * and it is hereby released to the general public under the following terms:
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted.
+ *
+ * https://github.com/mongodb/specifications/blob/master/source/auth/auth.md
+ * https://datatracker.ietf.org/doc/html/rfc7677
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_mongodb_scram_sha256;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_mongodb_scram_sha256);
+#else
+
+#include <string.h>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#include "arch.h"
+#include "misc.h"
+#include "memory.h"
+#include "common.h"
+#include "formats.h"
+#include "johnswap.h"
+#include "sha.h"
+#include "base64_convert.h"
+#include "hmac_sha.h"
+#include "simd-intrinsics.h"
+#include "pbkdf2_hmac_sha256.h"
+
+#if defined SIMD_COEF_32
+#define SIMD_KEYS		(SIMD_COEF_32 * SIMD_PARA_SHA256)
+#endif
+
+#define FORMAT_LABEL            "scram-pbkdf2-sha256"
+#define FORMAT_NAME             ""
+#define ALGORITHM_NAME          "SCRAM PBKDF2-SHA256 " SHA256_ALGORITHM_NAME
+#define PLAINTEXT_LENGTH        125
+#define HASH_LENGTH             44
+#define SALT_SIZE               sizeof(struct custom_salt)
+#define SALT_ALIGN              sizeof(uint32_t)
+#define BINARY_SIZE             32
+#define BINARY_ALIGN            sizeof(uint32_t)
+#define BENCHMARK_COMMENT       ""
+#define BENCHMARK_LENGTH        0x107
+#define FORMAT_TAG              "$scram-pbkdf2-sha256$"
+#define FORMAT_TAG_LENGTH       (sizeof(FORMAT_TAG) - 1)
+#define MAX_USERNAME_LENGTH     128
+
+#ifndef OMP_SCALE
+#define OMP_SCALE               1 // MKPC and scale tuned for i7
```

Comment:
Did you in fact do any tuning or is this a copy-paste leftover from the other format? Either way, these values don't look like they're optimal for either format.

Reply:
Oh, I see the previous values came from:

```
commit be422692c66478e59d0b614b34f2ee6cb84ed0de
Author: magnum <magnumripper@hushmail.com>
Date:   Tue Apr 9 22:35:14 2019 +0200

    OMP_SCALE & MKPC tuning, many formats, Closes #3091
```

So there was in fact some tuning of the related format, and it may not be that bad.